modules/services/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/SharingClient.kt

```diff
@@ -412,6 +469,16 @@ data class SharingRequest internal constructor(
         ) : Data {
             override fun toString() = "ClipVideo(title=${episode.title}, uuid=${episode.uuid}, start=${range.start.toSecondsWithSingleMilli()}, end=${range.end.toSecondsWithSingleMilli()})"
         }
+
+        class ReferralLink internal constructor(
+            private val referralCode: String,
+        ) : Data {
+            override val podcast: PodcastModel? = null
+
+            fun sharingUrl(host: String) = "$host/redeem-guest-pass/$referralCode"
```

Comment:
Not for now but if https://github.com/Automattic/pocket-casts-android/pull/2881#discussion_r1764370159 is added you'll be able to construct a link in a more cohesive way.